spark/src/main/scala/org/apache/spark/sql/delta/commands/WriteIntoDelta.scala

```diff
@@ -160,6 +160,13 @@ case class WriteIntoDelta(
     // it to true. This is unavoidable as getNewDomainMetadata uses information generated by
     // updateMetadata, so it needs to be run after that.
     val canUpdateMetadata = txn.canUpdateMetadata
+    if (txn.metadata.schemaString != null) {
+      // In cases other than CTAS (INSERT INTO, DataFrame write), block if values are provided for
+      // GENERATED ALWAYS AS IDENTITY columns.
+      IdentityColumn.blockExplicitIdentityColumnInsert(
+        txn.metadata.schema,
+        data.queryExecution.analyzed)
+    }
```

Comment:
Can you move this check up? Right now you're splitting the logic of when we use `canUpdateMetadata`. You can place it right after `finalSchema` is defined.